lib/livebook_web/live/settings_live.ex

```diff
@@ -314,10 +317,26 @@ defmodule LivebookWeb.SettingsLive do
 
   def handle_event("detach_file_system", %{"id" => file_system_id}, socket) do
     Livebook.Settings.remove_file_system(file_system_id)
+
+    case Ets.fetch(:file_system, "default_file_system") do
+      :error ->
+        ""
+
+      {:ok, default} ->
+        if default.file_system_id == file_system_id do
+          Ets.delete(:file_system, "default_file_system")
+        end
+    end
+
     file_systems = Livebook.Settings.file_systems()
     {:noreply, assign(socket, file_systems: file_systems)}
   end
 
+  def handle_event("make_default_file_system", %{"id" => file_system_id}, socket) do
+    Ets.insert(:file_system, "default_file_system", file_system_id: file_system_id)
```

Comment:
We should move this to `Livebook.Settings` :) There you can also see that we use a pluggable storage, rather than calling `Ets` directly, something like `storage().insert(:settings, "global", default_file_system_id: file_system_id)`.